File: backend/app/models/reservation.py
from sqlalchemy import Column, Integer, ForeignKey, DateTime, String
from sqlalchemy.orm import relationship
from app.db.base_class import Base

class Reservation(Base):
    __tablename__ = "reservations"
    
    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("user.id"))
    cancha_id = Column(Integer, ForeignKey("cancha.id"))
    reserva_time = Column(DateTime, nullable=False)
    start_time = Column(DateTime, nullable=False)
    end_time = Column(DateTime, nullable=False)
    estado = Column(String, index=True, nullable=False)

    user = relationship("User", back_populates="reservations")
    cancha = relationship("Cancha", back_populates="reservations")

File: backend/app/models/reservation.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class Reservation extends Model {
    static associate(models) {
      Reservation.belongsTo( models.User, {
        foreignKey: 'userId',
        as: 'user',
        onDelete: 'CASCADE',
     } );
     Reservation.belongsTo(models.Cancha, {
      foreignKey: 'canchaId',
      as: 'cancha',
      onDelete: 'CASCADE',
    });
    }
  }
  Reservation.init({
    fecha: DataTypes.DATEONLY,
    bloque: DataTypes.STRING,
    estado: DataTypes.STRING,
    confirmationToken: DataTypes.STRING
  }, {
    sequelize,
    modelName: 'Reservation',
  });
  return Reservation;
};